master/internal/api_trials.go

```diff
@@ -158,3 +158,52 @@ func (a *apiServer) KillTrial(
 	}
 	return &resp, err
 }
+
+func (a *apiServer) GetExperimentTrials(
+	_ context.Context, req *apiv1.GetExperimentTrialsRequest,
+) (*apiv1.GetExperimentTrialsResponse, error) {
+	resp := &apiv1.GetExperimentTrialsResponse{}
+
+	switch err := a.m.db.QueryProto("get_trials_for_experiment", &resp.Trials, req.ExperimentId); {
+	case err == db.ErrNotFound:
+		return nil, status.Errorf(codes.NotFound, "experiment %d not found:", req.ExperimentId)
+	case err != nil:
+		return nil, err
+	}
+	a.filter(&resp.Trials, func(i int) bool {
+		v := resp.Trials[i]
+		eliminate := false
+
+		if len(req.States) != 0 && !eliminate {
+			eliminate = true
+			for _, state := range req.States {
+				if state == v.State {
+					eliminate = false
+					break
+				}
+			}
+		}
+
+		return !eliminate
+	})
+
+	a.sort(resp.Trials, req.OrderBy, req.SortBy, apiv1.GetExperimentTrialsRequest_SORT_BY_ID)
+	if err := a.paginate(&resp.Pagination, &resp.Trials, req.Offset, req.Limit); err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (a *apiServer) GetTrial(_ context.Context, req *apiv1.GetTrialRequest) (
+	resp *apiv1.GetTrialResponse, err error,
+) {
+	// DISCUSS name query and var this way? y/n
```

Comment:
Not sure what other way to name it.. it's unfortunate, and it would've probably been nice to name them all like this, but it's too late now. I'd maybe go with `get_trial_proto`, or `proto_get_trial` so all the proto queries fall next to each other when viewing the files, ha.

Reply:
> I'd maybe go with get_trial_proto, or proto_get_trial so all the proto queries fall next to each other when viewing the files, ha.

let's do that